allure-rest-assured/src/test/java/io/qameta/allure/restassured/AllureRestAssuredTest.java

```diff
@@ -74,15 +76,22 @@ void shouldCreateAttachment(final List<String> attachmentNames, final AllureRest
     void shouldCatchAttachmentBody(final List<String> attachmentNames, final AllureRestAssured filter) {
         RestAssured.replaceFiltersWith(filter);
         final AllureResults results = execute();
-        final Attachment found = results.getTestResults().stream()
+
+        List<Attachment> actualAttachments = results.getTestResults().stream()
                 .map(TestResult::getAttachments)
-                .flatMap(Collection::stream)
-                .filter(attachment -> attachmentNames.contains(attachment.getName()))
-                .findAny()
+                .findFirst()
```

Comment:
I believe you may want to flatten all the attachments and collect them to list (w/o throwing a runtime exception), as you're iterating over the upstream results' collection. So when you pick only the first attachments' set, it may differ from the expected `attachmentNames`, which is not linked to particular results.

Reply:
You're right, it wasn't a good idea to get the first element of the attachment list. I've flattened the whole list attachment and I deleted the runtime exception, if there is no attachment already, the assertion will fail.